domains/pallets/messenger/src/fees.rs

```diff
@@ -0,0 +1,130 @@
+use crate::{BalanceOf, Config, Pallet, Relayers};
+use codec::{Decode, Encode};
+use frame_support::traits::ExistenceRequirement::AllowDeath;
+use frame_support::traits::{Currency, ExistenceRequirement, WithdrawReasons};
+use frame_support::PalletId;
+use scale_info::TypeInfo;
+use sp_runtime::traits::{AccountIdConversion, CheckedAdd, CheckedDiv, CheckedSub};
+use sp_runtime::{ArithmeticError, DispatchResult};
+
+/// Messenger Id used to store deposits and fees.
+const MESSENGER_PALLET_ID: PalletId = PalletId(*b"messengr");
+
+/// Execution Fee to execute a send or receive request.
+#[derive(Default, Debug, Encode, Decode, Clone, Copy, Eq, PartialEq, TypeInfo)]
+pub struct ExecutionFee<Balance> {
```

Comment:
Hmm, just a thought, something like `MessagerFee` looks more straightforward, whether it's a request or a response, the fee is charged as the cross-domain messages are delivered.

Reply:
I'm not sure TBH on the name side. MessengerFee doesn't fully explain the fee for relayers and compute. I'm gonna leave it as is for now